students/srepking/Lesson04/mailroom2.py

```diff
@@ -0,0 +1,122 @@
+def thank_you():
+    print('\n''Ok, you chose 1, lets see what we can do.')
+    while True:
+        name = input('\n''Type the full name to add entry, '
+                     '\'list\' to see list of names, or \'e\' to exit: > ')
+        # See if they want a list of names.
+        if name != 'list':
+            break
+        for x in donors:
+            print(x)
+    # Add Name to list if not in list
+    if name not in donors and name != 'e':
+        print('Name is not there, added to the list')
+        donors[name] = []
+    if name != 'e':
```

Comment:
i think it would be cleaner to first check for `e`, like
```
if name == 'e':
    return
```
and then do donor update or additional logic.

Reply:
Change made!

Reply:
👍